### PR TITLE
Wrote short description and demo of factory pattern

### PR DESCRIPTION
**Pull request by james-muldoon**
Information drawn from various sources, with no source being used enough to require explicit citation. Code samples are original.

### CODE REVIEW

File: src/pages/software-engineering/design-patterns/factory/index.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+The factory is a design pattern which aims to reduce the dependencies between classes instantiating objects and the objects themselves, by creating a "factory" class which handles the creation of classes. By doing this, we allow can allow subclasses to redefine which class to instantiate, and group potentially complex creation logic into a single interface.
```
Comment (bla-ke) [inline src/pages/software-engineering/design-patterns/factory/index.md]: Add before this line (1):
```
---
title: Factory
---
```

**Comment by amandeepmittal** [comment]
Hey @james-muldoon.

So I'd love to be able to merge your changes but it looks like you have some merge conflicts. ⚠️

Once you resolve these conflicts, I will be able to review your PR and merge it. 😊

**Comment by james-muldoon** [comment]
@amandeepmittal 
Hey Aman, I've done my best to merge the two and make it cohesive, let me know if there's anything else you need

**Comment by amandeepmittal** [comment]
Congratulations 🎉  on your first contribution @james-muldoon. Looks good to me. 👍

**Pull request merged**